python/tests/test_lowlevel.py

```diff
@@ -763,6 +763,40 @@ def test_table_extend_types(
         for i, expected_row in enumerate(expected_rows):
             assert table[len(table_copy) + i] == table_copy[expected_row]
 
+    @pytest.mark.parametrize("table_name", ["nodes"])
+    def test_table_keep_rows_errors(self, table_name, ts_fixture):
+        table = getattr(ts_fixture.tables, table_name)
+        n = len(table)
+        ll_table = table.ll_table
+        with pytest.raises(ValueError, match="must be of length"):
+            ll_table.keep_rows(np.ones(n - 1, dtype=bool))
+        with pytest.raises(ValueError, match="must be of length"):
+            ll_table.keep_rows(np.ones(n + 1, dtype=bool))
+        with pytest.raises(TypeError, match="Cannot cast"):
```

Comment:
This probably will be in the high-level tests, but given recent issues, probably worth adding tests for things like `False`.